app/src/org/commcare/android/database/DbHelper.java

```diff
@@ -103,6 +103,10 @@ public ContentValues getContentValues(Externalizable e) {
                 values.put(TableBuilder.scrubName(key), value);
             }
         }
+
+        if(blob.length > Math.pow(1024, 2)){
```

Comment:
Quick question: Do we get the 1MB issue when we put the record in the values cursor or write it? Feels dangerous to hard code the limit instead of catching the failing write and then throwing the RecordTooLargeException  to make it explicit to the code what is failing.